// @firebase/firestore/dist/memory/rn/index.js
import firebase from '@firebase/app';
import '@firebase/util';
import '@firebase/logger';
import '@firebase/webchannel-wrapper';
import { M as Ma, V as Vu, O, F as Fa, q as qa, U as Ua, K as Ka, z as za, J as Ja, H as Ha, X as Xa, Z as Za, e as eh, n as nh, B as Ba, y as yu, g as gu, a as Oa } from './prebuilt.rn-22709745.js';
import { Component } from '@firebase/component';

const name = "@firebase/firestore";
const version = "2.2.5";

/**
 * @license
 * Copyright 2017 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
const firestoreNamespace = {
    Firestore: Ma,
    GeoPoint: Vu,
    Timestamp: O,
    Blob: Fa,
    Transaction: qa,
    WriteBatch: Ua,
    DocumentReference: Ka,
    DocumentSnapshot: za,
    Query: Ja,
    QueryDocumentSnapshot: Ha,
    QuerySnapshot: Xa,
    CollectionReference: Za,
    FieldPath: eh,
    FieldValue: nh,
    setLogLevel: Ba,
    CACHE_SIZE_UNLIMITED: yu
};
/**
 * Configures Firestore as part of the Firebase SDK by calling registerService.
 *
 * @param firebase - The FirebaseNamespace to register Firestore with
 * @param firestoreFactory - A factory function that returns a new Firestore
 *    instance.
 */
function configureForFirebase(firebase, firestoreFactory) {
    firebase.INTERNAL.registerComponent(new Component('firestore', container => {
        const app = container.getProvider('app').getImmediate();
        return firestoreFactory(app, container.getProvider('auth-internal'));
    }, "PUBLIC" /* PUBLIC */).setServiceProps(Object.assign({}, firestoreNamespace)));
}

/**
 * @license
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * Registers the memory-only Firestore build with the components framework.
 */
function registerFirestore(instance) {
    configureForFirebase(instance, (app, auth) => new Ma(app, new gu(app, auth), new Oa()));
    instance.registerVersion(name, version);
}
registerFirestore(firebase);

export { registerFirestore };
